src/plugins/share/public/components/share_tabs.tsx

```diff
@@ -34,7 +34,8 @@ export const ShareMenuTabs = () => {
 
   tabs.push(linkTab);
 
-  if (shareMenuItems.length > 0) {
+  // do not show the export tab if the license is disabled
+  if (shareMenuItems.length > 0 && shareMenuItems[0].shareMenuItem?.disabled !== true) {
```

Comment:
This works when the first item in the array has the `disabled` field on the share menu item. That makes it a little hard to understand the intention behind the code, because I don't know if there is something special about the first item in the array. Maybe it would be more understandable if we look through the array and make sure all the items are enabled?

```
  const enabledItems = shareMenuItems.filter(({ shareMenuItem }) => !shareMenuItem?.disabled);
  // show the export tab if all share menu items are enabled
  if (shareMenuItems.length > 0 && enabledItems.length === shareMenuItems.length) {
    tabs.push(exportTab);
  }
```

Then, we could use this mock data in a test:
```
    const testItems = [
      { shareMenuItem: { name: 'test', disabled: false }, label: CSV, generateExport: jest.fn() },
      { shareMenuItem: { name: 'test', disabled: true }, label: CSV, generateExport: jest.fn() },
    ];
```

Reply:
Ah good point, I was thinking if there is a shareMenuItem at all then we just need to check the first value to see if it has a disabled license or not. But CSV reports don't require the same licensing standard right? So it definitely makes sense to have to loop through to check, thanks!